src/core/exception.d

```diff
@@ -574,14 +574,20 @@ extern (C) void onInvalidMemoryOperationError(void* pretend_sideffect = null) @t
  */
 extern (C) void onSwitchError( string file = __FILE__, size_t line = __LINE__ ) @safe pure nothrow
 {
-    throw new SwitchError( file, line, null );
+    version (D_Exceptions)
```

Comment:
I suggest adding a comment here that this is temporary, and that once we get either a `pure` malloc, or an `alloca` for -betterC that these should be changed to an `assert` for both cases.  Either that, or file a bugzilla issue with the same.